### PR TITLE
qpdf: changed build_style to cmake.

### PR DESCRIPTION
**Pull request by benalb**
#### Testing the changes
- I tested the changes in this PR: **YES**

 
#### Local build testing
- I built this PR locally for my native architecture, (X86-64-LIBC)

### CODE REVIEW

File: srcpkgs/qpdf/template [hunk 1]
```diff
@@ -1,45 +1,38 @@
...
 distfiles="${homepage}/archive/release-qpdf-${version}.tar.gz"
-checksum=64bbb654ec19b6812284ef1bc40c57b091de5281f265f1c3c83dd8ccf6bedcf5
+checksum=729112f37d5eb53060cc8f5c4346e814670f147cb8498d47d010415754ce8087
 
 if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
-	makedepends+=" libatomic-devel"
-	LDFLAGS+=" -latomic"
+        makedepends+=" libatomic-devel"
+        LDFLAGS+=" -latomic"
 fi
```
Comment (TinfoilSubmarine) [inline srcpkgs/qpdf/template:21]: Need to patch out tests if CROSS_BUILD:
```
post_patch() {
        if [ "$CROSS_BUILD" ]; then
                vsed -e "/\(fmt_lld\|fmt_i64d\|fmt_i64lld\)/d" -i libqpdf/CMakeLists.txt
        fi
}
```

**Comment by TinfoilSubmarine** [comment]
Needs [this patch](https://gist.github.com/TinfoilSubmarine/82528118535a026162163e6b17a26faa) to cross-build successfully, need to test if this produces a valid, working binary though.

File: srcpkgs/qpdf/template (hunk 1 above)
Comment (tibequadorian) [reply srcpkgs/qpdf/template:21]: No need to patch. Works too with `configure_args=-DLL_FMT=%lld -DRANDOM_DEVICE=/dev/urandom`

Comment (TinfoilSubmarine) [reply srcpkgs/qpdf/template:21]: Ah, shows how much I know my cmake-foo. Thanks!

File: srcpkgs/qpdf/template [hunk 2]
```diff
@@ -1,35 +1,32 @@
...
 license="Apache-2.0"
 homepage="https://github.com/qpdf/qpdf"
 changelog="https://raw.githubusercontent.com/qpdf/qpdf/stable/manual/release-notes.rst"
 distfiles="${homepage}/archive/release-qpdf-${version}.tar.gz"
-checksum=64bbb654ec19b6812284ef1bc40c57b091de5281f265f1c3c83dd8ccf6bedcf5
+checksum=729112f37d5eb53060cc8f5c4346e814670f147cb8498d47d010415754ce8087
 
 if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
 	makedepends+=" libatomic-devel"
 	LDFLAGS+=" -latomic"
```
Comment (tibequadorian) [inline srcpkgs/qpdf/template:19]:
```suggestion
	LIBS="-latomic"
```
to fix armv6l

Comment (tibequadorian) [reply srcpkgs/qpdf/template]: only change this line as i suggested

**Comment by benalb** [comment]
Builds OK, I have tested it in X86_64, i686 and armv7l. I will wait till upstream realeses 10.6.4 to push the PR.
Thank you very much @tibequadorian and @TinfoilSubmarine

File: srcpkgs/qpdf/template (hunk 2 above)
Comment (tibequadorian) [reply srcpkgs/qpdf/template]: Maybe "to fix armv6l" was misleading, please change the line as I suggested.

**Comment by Piraty** [comment]
is that worth it?

**Comment by benalb** [comment]
It's an upstream [decission](https://github.com/qpdf/qpdf/releases/tag/release-qpdf-10.6.3.0cmake1)

File: srcpkgs/qpdf/template (hunk 2 above)
Comment (tibequadorian) [reply srcpkgs/qpdf/template]: @benalb see above

**Comment by benalb** [comment]
as there is a soname bump involved in the update to 11.0.0, I will close this WIP and create a new PR.

**Pull request closed without merging**